Entities/Common/Movement/FallDamage.as

```diff
@@ -33,24 +33,29 @@ void onCollision(CBlob@ this, CBlob@ blob, bool solid, Vec2f normal, Vec2f point
 	{
 		bool doknockdown = true;
 
-		if (damage > 0.0f)
-		{
-			// check if we aren't touching a trampoline
-			CBlob@[] overlapping;
-
-			if (this.getOverlapping(@overlapping))
+		// better check for trampolines
+		CBlob@[] blobs_around;
+		if (getMap().getBlobsInRadius(this.getPosition(), this.getRadius() * 4, blobs_around)) {
+			for (uint i = 0; i < blobs_around.length; i++)
 			{
-				for (uint i = 0; i < overlapping.length; i++)
-				{
-					CBlob@ b = overlapping[i];
-
-					if (b.hasTag("no falldamage"))
-					{
-						return;
-					}
-				}
+				CBlob@ b = blobs_around[i];
+
+				if (b is null) continue;
```

Comment:
You wont need a null check here, all the blobs returned by getBlobsInRadius are valid (and you are not storing it over multiple ticks)